### PR TITLE
Yearn assets base

### PR DESCRIPTION
**Pull request by hudeintres**

### CODE REVIEW

File: package.json
Comment (clothic) [inline package.json:1]: This is incredibly weird. why did you add all the packages like this? Here is an example. DO NOT MERGE UNTIL THIS IS ADDRESSED
```
"node_modules/@ampproject/remapping": {
      "version": "2.2.0",
      "dev": true,
      "license": "Apache-2.0",
      "dependencies": {
        "@jridgewell/gen-mapping": "^0.1.0",
        "@jridgewell/trace-mapping": "^0.3.9"
      },
      "engines": {
        "node": ">=6.0.0"
      }
    },
```

**Review by clothic — CHANGES_REQUESTED** [review]

File: packages/contracts/markets/base/reservesConfigs.ts [hunk 1]
```diff
@@ -179,6 +195,21 @@ export const strategyVeloWETHDAI: IReserveParams = {
...
+export const strategyVelowstETHWETH: IReserveParams = {
+  strategy: rateStrategyUnborrowable,
+  baseLTVAsCollateral: ethers.utils.parseUnits('0.65',18).toString(), 
+  liquidationThreshold: ethers.utils.parseUnits('0.7',18).toString(),
+  liquidationBonus: ethers.utils.parseUnits('1.1',18).toString(), 
+  borrowingEnabled: false,
+  reserveDecimals: '18', 
+  aTokenImpl: eContractid.AToken,
+  assetType: 5, //5 is enum for velo
+  supplyCap: "130",
```
Comment (clothic) [inline packages/contracts/markets/base/reservesConfigs.ts:207]: why allow unlimited supplying of the other ones but not this one?

File: packages/contracts/markets/base/reservesConfigs.ts [hunk 2]
```diff
@@ -179,6 +195,21 @@ export const strategyVeloWETHDAI: IReserveParams = {
...
+  borrowCap: '0',
+  borrowFactor: ethers.utils.parseUnits('1',18).toString(), //100% for now
```
Comment (clothic) [inline packages/contracts/markets/base/reservesConfigs.ts:209]: Does borrow factor matter when borrowing this asset is disabled?

also remind me, 100% means there is no extra collateral than normal needed when trying to borrow?

**Review by clothic — COMMENTED** [review]
Overall, looks good. Small comments for clarity and cleanup. The previous comment is the most important one

File: package.json
Comment (hudeintres) [reply package.json:1]: you're right, I will remove this

File: packages/contracts/markets/base/commons.ts [hunk 3]
```diff
@@ -206,6 +210,10 @@ export const CommonsConfig: ICommonConfiguration = {
         address: "0xe9c898ba654dec2ba440392028d2e7a194e6dc3e",
         type: 4,
       },
+      "vAMM-wstETH/WETH": {
+        address: "0xdf7c8f17ab7d47702a4a4b6d951d2a4c90f99bf4",
+        type: 2,
+      },
```
[the remark this answers is not in the document]
Comment (hudeintres) [reply packages/contracts/markets/base/commons.ts:216]: the others are yearn tokens which don't have staking

File: packages/contracts/markets/base/index.ts [hunk 4]
```diff
@@ -48,6 +54,15 @@ export const BaseConfig: IBaseL2Configuration = {
     "cbETH-WETH-BPT": strategyCbETHBPT,
     "rETH-WETH-BPT": strategyrETHETHBPT,
     "cbETHWETHCRV": strategycbETHWETHCRV,
+
+    wstETH: strategywstETH,
+    "vAMM-wstETH/WETH": strategyVelowstETHWETH,
```
[the remark this answers is not in the document]
Comment (hudeintres) [reply packages/contracts/markets/base/index.ts:59]: the base ones that don't need a string don't have them. We need strings for names with weird symbols like - and /

File: packages/contracts/markets/base/index.ts [hunk 5]
```diff
@@ -48,6 +54,15 @@ export const BaseConfig: IBaseL2Configuration = {
...
+    
+    "yv-vAMM-cbETH/WETH": strategyyvVelocbETHWETH,
+    "yv-vAMM-WETH/USDbC": strategyyvVeloWETHUSDbC,
+    // "yv-vAMM-wstETH/WETH": ""
```
[the remark this answers is not in the document]
Comment (hudeintres) [reply packages/contracts/markets/base/index.ts:63]: keep this here, I will add this one soon. It wasn't added in the original iteration since yearn forgot to deploy it

File: packages/contracts/markets/base/reservesConfigs.ts [hunk 6]
```diff
@@ -104,6 +104,22 @@ export const strategyCbETH: IReserveParams = {
...
+export const strategywstETH: IReserveParams = {
+  strategy: rateStrategyCbETH,
+  baseLTVAsCollateral: ethers.utils.parseUnits('0.71',18).toString(),
+  liquidationThreshold: ethers.utils.parseUnits('0.76',18).toString(),
+  liquidationBonus: ethers.utils.parseUnits('1.06',18).toString(),
+  borrowingEnabled: true,
+  reserveDecimals: '18',
+  aTokenImpl: eContractid.AToken,
+  assetType: 8, 
+  supplyCap: '4000', //4,000
```
[the remark this answers is not in the document]
Comment (hudeintres) [reply packages/contracts/markets/base/reservesConfigs.ts:116]: they are set as 1/10 of LP total supply. The value I set here is then multiplied by the number of decimals in the lp token

File: packages/contracts/markets/base/reservesConfigs.ts (hunk 1 above)
Comment (hudeintres) [reply packages/contracts/markets/base/reservesConfigs.ts:207]: this one is a aerodrome LP token. These need supply caps. Yearn tokens don't need supply caps. The other ones I think u mean the yearn tokens

File: packages/contracts/markets/base/reservesConfigs.ts (hunk 2 above)
Comment (hudeintres) [reply packages/contracts/markets/base/reservesConfigs.ts:209]: no borrow factor does not matter. And yes you're right

File: packages/contracts/markets/base/reservesConfigs.ts (hunk 6 above)
[the remark this answers is not in the document]
Comment (clothic) [reply packages/contracts/markets/base/reservesConfigs.ts:116]: Ok what about 4000?

Does this mean the max supplied in a single deposit can be 

4000* numDecimals tokens?

File: packages/contracts/lib/solmate
Comment (clothic) [inline packages/contracts/lib/solmate:1]: Is this file pushed to git?

**Review by clothic — APPROVED** [review]

File: packages/contracts/markets/base/reservesConfigs.ts (hunk 6 above)
[the remark this answers is not in the document]
Comment (hudeintres) [reply packages/contracts/markets/base/reservesConfigs.ts:116]: yes, 4000 * numDecimals of wei can be deposited. In other words, 4000 tokens can be deposited

File: packages/contracts/lib/solmate
Comment (hudeintres) [reply packages/contracts/lib/solmate:1]: yep, forge handles packages as submodules.
